contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.avro;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.avro.AvroRuntimeException;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.lib.io.fs.AbstractFileInputOperator;
+
+/**
+ * <p>
+ * Avro File Input Operator
+ * </p>
+ * A specific implementation of the AbstractFileInputOperator to read Avro
+ * container files.<br>
+ * No need to provide schema,its inferred from the file<br>
+ * Users can add the {@link IdempotentStorageManager.FSIdempotentStorageManager}
+ * to ensure exactly once semantics with a HDFS backed WAL.
+ * 
+ * @displayName AvroFileInputOperator
+ * @category Input
+ * @tags fs, file,avro, input operator
+ * @since 3.3.0
+ */
+@InterfaceStability.Evolving
+public class AvroFileInputOperator extends AbstractFileInputOperator<GenericRecord>
+{
+
+  private transient long offset = 0L;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  private transient DataFileStream<GenericRecord> avroDataStream;
+  private transient GenericRecord record = null;
+
+  public final transient DefaultOutputPort<GenericRecord> output = new DefaultOutputPort<GenericRecord>();
+
+  @OutputPortFieldAnnotation(optional = true)
+  public final transient DefaultOutputPort<String> completedFilesPort = new DefaultOutputPort<String>();
+
+  @OutputPortFieldAnnotation(optional = true)
+  public final transient DefaultOutputPort<String> errorRecordsPort = new DefaultOutputPort<String>();
+
+  @Override
```

Comment:
Well yes it should be there too. But considering the review is for this file I suggest to put it here because the default functionality is overridden or extended.